### ISSUE TITLE
VS Code Razor syntax highlighting incorrect 

### ISSUE THREAD

**Issue by TrieBr** [dotnet/aspnetcore#32155]
### Describe the bug
I'm posting this based on aspnet/Razor.VSCode pointing to this repository. Apologies if its the wrong location.

I recently stared using VSCode instead of Visual studio and noticed the razor syntax highlighting is sub optimal. In one particular case, it was highlighting content totally different of how the compiler would interpret it, leading to subtle bugs.

This particular case involves the @ symbol that precedes code blocks and statements in razor templates. 

In the following case, this if statement is not actually compiled as an if statement, yet Razor syntax highlighting in VS Code shows it as an if statement instead of plaintext. 

![image](https://user-images.githubusercontent.com/348719/116001114-d06f2b00-a5a7-11eb-81cd-2c2869211780.png)

Here is the same snippet in Visual Studio 2019. Syntax highlighting is correctly marking it as plaintext.

![image](https://user-images.githubusercontent.com/348719/116001068-9736bb00-a5a7-11eb-9853-a9ce93b3140c.png)

### To Reproduce

1. Install the C# Extension for VS Code (I'm using v1.23.11).
2. Create a new file 
3. Select a language `aspnetcorerazor` (Do not select `Razor`, that one seems to work correctly, but is missing some blazor stuff)
4. Add some minimal code:
```
<div>
if (true) {


}
</div>
```
5. Observe that the if statement is highlighted as if it was a code block when in reality it is compiled as plaintext.

### Further technical details
- C# Extension for VS Code v1.23.11
- ASP.NET Core version: 5.0.201
- Include the output of `dotnet --info`
```
.NET SDK (reflecting any global.json):
 Version:   5.0.201
 Commit:    a09bd5c86c

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.19042
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\5.0.201\

Host (useful for support):
  Version: 5.0.4
  Commit:  f27d337295

.NET SDKs installed:
  5.0.101 [C:\Program Files\dotnet\sdk]
  5.0.201 [C:\Program Files\dotnet\sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.All 2.1.26 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.App 2.1.26 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.1.13 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 5.0.4 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 2.1.26 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.13 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 5.0.4 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.WindowsDesktop.App 3.1.13 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 5.0.4 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]

To install additional .NET runtimes or SDKs:
  https://aka.ms/dotnet-download
```

- VS Code:
```
Version: 1.55.2 (user setup)
Commit: 3c4e3df9e89829dce27b7b5c24508306b151f30d
Date: 2021-04-13T09:35:57.887Z
Electron: 11.3.0
Chrome: 87.0.4280.141
Node.js: 12.18.3
V8: 8.7.220.31-electron.0
OS: Windows_NT x64 10.0.19042
```

**Comment by hammypants** [comment]
Yeah, this kinda thing is all over the ASP.NET Core Razor stuff. It's largely unusable.

**Comment by mikesigs** [comment]
I am getting started with Blazor this weekend and I'm also finding the Razor highlighting in VS Code to be really bad. It's difficult to work on code when your editor is telling you something is wrong, but the compiler is telling you it's right.

Here are a few examples that all compile and run properly, but show a false positive error with RED highlighting: 

### Inline Ternary Expression
![image](https://user-images.githubusercontent.com/811177/133003862-db8c44f3-8388-4b0a-96ab-3c476093bb58.png)

### Inline Lambda Expression
![image](https://user-images.githubusercontent.com/811177/133003883-0d6094c1-a732-48a9-8f70-42a25d5fade8.png)

It formats and runs properly without the quotes.
![image](https://user-images.githubusercontent.com/811177/133003895-879f2b5d-afde-4c3d-a081-18e9fe44762e.png)

### Quoted Handler Function
![image](https://user-images.githubusercontent.com/811177/133003946-a961c275-665c-4c88-9d3c-375343b44138.png)

Again, this formats and runs properly without the quotes.
![image](https://user-images.githubusercontent.com/811177/133003958-8943966e-26c6-4395-8451-56e177f546d9.png)
